Bug 1196569 - Private Browsing product page

malena opened the pull request:
https://bugzilla.mozilla.org/show_bug.cgi?id=1196569

flodolo commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,136 @@
+
+{% extends "/firefox/base-resp.html" %}
+
+{% block page_title %}{{ _('Keep your Firefox in sync') }}{% endblock %}
```
Need to update the page title.

flodolo commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,136 @@
+          <br>
+          {{ _('The most privacy.') }}
+          <br>
+          {{ _('Only from Firefox.') }}
```
We have already [exposed](More protection. <br>The most privacy. <br>Only from Firefox.) a similar string for the campaign landing page, I'd suggest to reuse the same
`More protection. <br>The most privacy. <br>Only from Firefox.`

sgarrity replied:
@flodolo To be clear here, it should change in this case because we already have that string in the landing page, but usually this would be a better way to wrap the strings, right?

flodolo replied:
Yes and no. This is not horrible for localization, but it kind of forces you to have 3 sentences on 3 lines, when maybe you want to try something different.

ckprice commented:
>Need to update the page title.

Sorry guys, I need to get this from Matej. Will have today.

ckprice commented:
@flodolo @malena copy document updated with title and desc.

[TITLE]

Firefox Web browser — Private Browsing with Tracking Protection — Mozilla


[DESCRIPTION]

Get more protection and the most privacy, only from Firefox. No other browser gives you so much control over your digital footprint.

flodolo commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,136 @@
+      <p>
+        {{ _('When you see the shield to the left of the address bar, that means Firefox is protecting you from potential trackers.') }}
+        {{ _('You can disable Tracking Protection for a single site, or turn it off entirely.') }}
+        {{ _('It’s up to you.') }}
```
Please keep this two sentences together
```
{{ _('You can disable Tracking Protection for a single site, or turn it off entirely. It’s up to you.') }}
```

flodolo commented:
Besides a couple of notes on strings, I can't see anything wrong from a quick text. Waiting to double check on next push.

malena commented:
@flodolo Updated strings.

flodolo commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,130 @@
+
+{% extends "/firefox/base-resp.html" %}
+
+{% block page_title %}{{ _('Firefox Web browser — Private Browsing with Tracking Protection — Mozilla ') }}{% endblock %}
```
Please remove the trailing space.

flodolo commented:
Strings extracted locally, please warn if anything changes. Plan to expose them within 24 hours.

ckprice commented:
@flodolo we have no new strings aside from the Title and Desc were added to the [copy document](https://docs.google.com/document/d/15Egy67jk_eD1LqFoWGGzh-nMXH1t2iEpV3K0VkTMDQE/edit) yesterday.

I haven't stepped through this PR, but if all the strings in the copy doc are in, then we are good to go here.

flodolo commented:
@ckprice Thanks, luckily this PR had the correct title and desc from the start (just double checked).

malena commented:
Yep I added the correct title and desc as soon as it became available in the doc. I am still working on layout/design for now. @flodolo let me know when I should push this to demo server.

flodolo commented:
Note: the page is available on demo5 only with the /b redirect
https://www-demo5.allizom.org/b/firefox/private-browsing

schalkneethling commented on etc/httpd/global.conf:
```diff
@@ -285,6 +285,9 @@ RewriteRule ^/en-US/firefox/channel/android(/?)$ /b/en-US/firefox/channel/androi
 RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?firefox/tiles(.*)$ /b/$1firefox/tiles$2 [PT]
 
+# bug 1196569 
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)/firefox/private-browsing(.*)$ /b/$1firefox/private-browsing$2 [PT]
```
I reckon you need something like this:

```
# bug 1205357
RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?firefox/choose(/?)$ /b/$1firefox/choose$2 [PT]
```

schalkneethling replied:
Reckon `(.*)` is causing not to?

jpetto commented:
@malena - Just a heads up that some additional GTM requests have come through in [bug 1217883](https://bugzilla.mozilla.org/show_bug.cgi?id=1217883).

malena commented:
Okay thanks! @jpetto

flodolo commented:
Nit found by a localizer: small#tracking-protection-caveat takes the entire width of the page, small#control-center-caveat doesn't (37%).

malena commented:
@flodolo Ahh okay yes the PS design has the caveat for small#control-center-caveat break at a smaller width hence my css width rule, but since this page is localized it would be better to maintain the same wider width. Thanks!

malena commented:
r?

flodolo commented:
"Try Private Browsing" is not hard-coded
https://bugzilla.mozilla.org/show_bug.cgi?id=1218543

I remember checking this page and not noticing this issue, we have this string [extracted](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/private-browsing.lang#L20), when did it change?

craigcook commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,170 @@
+        <h1>{{ _('More protection. <br>The most privacy. <br>Only from Firefox.') }}</h1>
+        <div class="download-buttons">
+          <div class="firefox-latest-cta">
+            <a href="https://www.mozilla.org" class="button-flat-dark" data-highlight="privateWindow" role="button"> Try Private Browsing </a>
```
Unwrapped string, "Try private browsing"

malena replied:
Oooh yes, there was some confusion with what text was originally in there for that line. The copy had a different sentence that appeared after l10n. So during that confusion I left it like that to remember to check if it. @flodolo Don't worry, the copy hasn't changed! I made sure of that!

craigcook commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,170 @@
+      <div class="download-buttons">
+        <div class="firefox-latest-cta">
+          <h2>{{ _('Try it now to stay in control') }}</h2>
+          <a href="https://www.mozilla.org" class="button-flat-dark" data-highlight="privateWindow" role="button"> Try Private Browsing </a>
```
Unwrapped string, "Try private browsing"

craigcook commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,170 @@
+        <strong>
+          <a class="android-link" href="https://support.mozilla.org/en-US/kb/private-browsing-firefox-android">{{ _('Learn how to use Private Browsing') }}</a>
+          <a class="ios-link" href="https://support.mozilla.org/kb/private-browsing-firefox-ios
+">{{ _('Learn how to use Private Browsing') }}</a>
```
Nit: unnecessary line break here

flodolo commented:
Can you push an update to demo server? That's important for testing and check how the button looks.

schalkneethling commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,169 @@
+        <h1>{{ _('More protection. <br>The most privacy. <br>Only from Firefox.') }}</h1>
+        <div class="download-buttons">
+          <div class="firefox-latest-cta">
+            <a href="https://www.mozilla.org" class="button-flat-dark" data-highlight="privateWindow" role="button">{{ _('Try Private Browsing') }}</a>
```
You can use the url helper here:
`{{ url('mozorg.home') }}

alexgibson replied:
This URL should probably fall back to something more useful than the homepage. If someone doesn't have the `privateWindow` target in their pallet, then the button will fall back to a regular link instead of highlighting the icon. A SUMO article like https://support.mozilla.org/kb/private-browsing-use-firefox-without-history/may be more appropriate.

schalkneethling commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,169 @@
+            </div>
+          </div>
+          <div class="firefox-old-cta">
+            <p>{{ _('Your Firefox is not up to date. Download the latest version to use the new Private Browsing.') }}</p>
```
Not sure if it matters, and probably to late for a string change but, seems a little strange that this string does include the "with Tracking Protection" part?

malena replied:
The copy has been reviewed and approved. Too late to make any changes.

schalkneethling replied:
:+1:

schalkneethling commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,169 @@
+      </p>
+      <p class="footer">
+        <strong>
+          <a class="android-link" href="https://support.mozilla.org/kb/mixed-content-blocker-firefox-android">{{ _('Learn more about the Control Center') }}</a>
```
This link says "Control Center" but the caveat states that Fx on Android does not have a control center?

schalkneethling commented on bedrock/firefox/templates/firefox/private-browsing.html:
```diff
@@ -0,0 +1,169 @@
+      <div class="download-buttons">
+        <div class="firefox-latest-cta">
+          <h2>{{ _('Try it now to stay in control') }}</h2>
+          <a href="https://www.mozilla.org" class="button-flat-dark" data-highlight="privateWindow" role="button">{{ _('Try Private Browsing') }}</a>
```
You can use the url helper here as well.

alexgibson replied:
This one should be a more appropriate fallback link too, like the other CTA above :+1:

schalkneethling commented on media/css/firefox/private_browsing/private-browsing-conditionals.less:
```diff
@@ -0,0 +1,170 @@
+@import "../../sandstone/lib.less";
+
+.download-buttons  {
+    > div {
```
Nit: No need to nest, can just be `.download-buttons > div`

schalkneethling commented on media/css/firefox/private_browsing/private-browsing-conditionals.less:
```diff
@@ -0,0 +1,170 @@
+        display: none;
+    }
+    .download-buttons {
+    }
```
Empty rule?

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+
+#wrapper {
+    border-top: 2px solid #fff;
+    background-color: #0095dd;
```
Should this be @mozIDBlue https://github.com/mozilla/bedrock/blob/master/media/css/sandstone/lib.less#L66

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+    color: @mozIDGrey;
+    .open-sans-light();
+    .font-size(20px);
+
```
Nit: Superfluous empty line.

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+    a {
+        display: none;
+    }
+    a::after {
```
Use a single `:` which will that ensure IE8 can join the party :tada: :)

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+        display: none;
+    }
+    a::after {
+        content:" »";
```
Nit: Use single quotes

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+        padding: 0 30px;
+    }
+    p {
+        margin: @baseLine auto 0 auto;
```
Nit: You can leave off the last `auto`

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+    header {
+        width: 490px;
+        padding-top: 80px;
+        float: left;
```
Nit: Would prefer if the order was `float`, `padding-top`, `width`

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+    }
+    .blurb {
+        background: url('/media/img/firefox/private-browsing/sprite-tracking-protection.png') no-repeat;
+        background: linear-gradient(transparent, transparent), url('/media/img/firefox/private-browsing/sprite-tracking-protection.svg');
```
Nit: Use gradient mixin? https://github.com/mozilla/bedrock/blob/master/media/css/sandstone/lib.less#L290

malena replied:
@schalkneethling  Hmm that's only there for IE fallback, not really a gradient. Wondering if using a mixin would make it hard to understand what's happening here? I can make a comment perhaps? What do you think?

schalkneethling replied:
depends on which IE version(s) this is for. It is only supported by IE10+ and the mixin adds the `-ms-filter` part for older versions of IE that does not support `linear-gradient`.

http://caniuse.com/#search=linear-gradient

schalkneethling replied:
With that said IE looks good in general so, maybe it is not an issue at all.

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+        background-position: 135px 0;
+        padding-right: 24px;
+    }
+    .private-browsing::after {
```
Single `:` here as well

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+    }
+    .private-browsing::after {
+        color: #6fbf4a;
+        content: "+";
```
Single quotes

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+       
+        p {
+            background: url('/media/img/firefox/private-browsing/sprite-features.png');
+            background: linear-gradient(transparent, transparent), url('/media/img/firefox/private-browsing/sprite-features.svg');
```
Gradient mixin?

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+
+.control-center {
+    background: url('/media/img/firefox/private-browsing/control-center.png');
+    background: linear-gradient(transparent, transparent), url('/media/img/firefox/private-browsing/control-center.svg');
```
And here

schalkneethling commented on media/css/firefox/private_browsing/private-browsing.less:
```diff
@@ -0,0 +1,489 @@
+        .font-size(25px);
+    }
+    .footer a::after {
+        content:" »";
```
Single quotes and single `:` above

schalkneethling replied:
Nit: Space after `content:`

schalkneethling commented on media/js/firefox/private-browsing.js:
```diff
@@ -0,0 +1,47 @@
+        $html.addClass('non-firefox');
+    }
+
+})(window.jQuery, window.Modernizr, window.site);
```
Nit: You can shorten this to `jQuery, Modernizr, site`, `window` is implied. Although, do you need Modernizr?

malena replied:
When I remove the window reference, window.isFirefox() won't work.  Also I thought we used Modernizr on all pages? Especially this one which needs to work in IE since we are using HTML5 elements.

alexgibson replied:
Fwiw, I think it's ok to keep these as `window.jQuery` etc, as it makes it obvious they are global. As for Modernizr, this should not be required unless you are using any of its features specifically. We include an [HTML5 Shiv](https://github.com/mozilla/bedrock/blob/master/bedrock/base/templates/base-resp.html#L45-L48) is all pages by default for IE.

malena replied:
Ahh okay great @alexgibson. I just try to keep my page consistent with others and I actually just copied this from another page. Will remove it.

schalkneethling commented on media/js/firefox/private-browsing.js:
```diff
@@ -0,0 +1,47 @@
+;(function($, Modernizr, site) {
+    'use strict';
+
+    var $html = $(document.documentElement);
```
Nit: You can shorten this to `$('html');`

malena replied:
Ahh yes, but actually I've seen $(document.documentElement); on other pages. Wondering why that has been used over $('html')?

schalkneethling commented:
Not sure if it is only on this page (I reckon there may be others) but, the color contrast of the secondary links under the download button is not good at all. In fact, the text is almost illegible.

![screen shot 2015-10-27 at 12 29 20](https://cloud.githubusercontent.com/assets/10350960/10755433/83fbc7fe-7ca6-11e5-908a-fe25658a50ea.png)

schalkneethling commented:
The license text in the footer under the wordmark is effected by the paragraph rule here:
https://github.com/mozilla/bedrock/pull/3391/files#diff-17c14532df4ae71b74268ce01e6aba9aR49

causing the font size to be to big. Perhaps add a body (or a class on `main`) class such as `private-browsing` and then wrap the `p` and the header rules inside it.

![screen shot 2015-10-27 at 12 40 41](https://cloud.githubusercontent.com/assets/10350960/10755819/c28a1884-7ca8-11e5-85b8-a6ce19c0631e.png)

schalkneethling commented on media/css/firefox/private_browsing/private-browsing-conditionals.less:
```diff
@@ -0,0 +1,170 @@
+        h2 {
+            .font-size(30px);
+            line-height: 1.4;
+            width: 440px;
```
I do not believe you need to make the width smaller here. It can just use the `500px` from `.cta h2`. Making it smaller causes a disconnect between the message and the button I feel.

Currently:

![screen shot 2015-10-27 at 12 52 25](https://cloud.githubusercontent.com/assets/10350960/10755973/a9e11598-7ca9-11e5-8c4d-19fc62a9f4fa.png)

As apposed to:

![screen shot 2015-10-27 at 12 52 38](https://cloud.githubusercontent.com/assets/10350960/10755980/b286c0da-7ca9-11e5-91e1-3c0be806305e.png)

malena replied:
The design had the sentence break specifically in those spots for desktop. Not sure if I should be changing that? I can push current to demo and send link to design for confirmation on this.

schalkneethling replied:
Perhaps have them take a look, not a blocker. Thx @malena

schalkneethling replied:
This bug confirms that I was wrong https://bugzilla.mozilla.org/show_bug.cgi?id=1218848

schalkneethling commented:
Alignment problems when showing the Google Play download button:

![screen shot 2015-10-27 at 12 59 56](https://cloud.githubusercontent.com/assets/10350960/10756127/bb354480-7caa-11e5-848a-e5b2439cd118.png)

And just above the footer:

![screen shot 2015-10-27 at 13 00 04](https://cloud.githubusercontent.com/assets/10350960/10756137/c5934f26-7caa-11e5-9aff-bae49895d930.png)

schalkneethling commented:
Possibly also on ios (although in the ios simulator it looked fine) - This is replacing for example android, with ios on the `html` element.

![screen shot 2015-10-27 at 13 02 16](https://cloud.githubusercontent.com/assets/10350960/10756191/19c23ea4-7cab-11e5-9e7c-020e09627440.png)

And above the footer:

![screen shot 2015-10-27 at 13 02 25](https://cloud.githubusercontent.com/assets/10350960/10756199/2159d438-7cab-11e5-91e5-835e14bdd71c.png)

alexgibson commented on media/js/firefox/private-browsing.js:
```diff
@@ -0,0 +1,47 @@
+
+    Mozilla.HighlightTarget.init('.button-flat-dark');
+
+    if (window.isFirefox()) {
```
Just a side note: this will return true for Firefox for Android as well as Desktop. Perhaps this is intentional, but it's worth pointing out that the UITour functionality will only work on the desktop browser. I'm not sure what Firefox for Android users should see here? (sorry for the drive-by comment).

malena replied:
Ohh okay. Firefox on Android should see what Chrome on Android sees. So I would need to add another check to see if mobile too.

schalkneethling commented:
IE is being a ... on smaller viewports (IE9)

![windows 7 - ie9](https://cloud.githubusercontent.com/assets/10350960/10756453/e4ca2372-7cac-11e5-8920-d7eb52443f79.png)

and

![windows 7 - ie92](https://cloud.githubusercontent.com/assets/10350960/10756470/ed644f80-7cac-11e5-9cc8-f8ba1d01b75a.png)

malena commented:
@schalkneethling I've tackled the ie9 layout issues. Also learned ie9 does not support top:inherit. :confused:

schalkneethling commented:
IE :angry:

malena commented:
@schalkneethling I've updated the font colour of that text that wasn't very contrasty. I also updated alignment issues on mobile. Will be doing another round of tests once demo5 is updated but for now this is ready for another pass when you have a chance. Thanks!

schalkneethling commented:
When the `html` element class is set to `ios js non-firefox loaded` you will see in the screen shot below, that the critters are causing horizontal scrolling at small viewports:

![screen shot 2015-10-28 at 16 58 17](https://cloud.githubusercontent.com/assets/10350960/10792451/6d0dd89e-7d95-11e5-868e-4f32cab1f5fa.png)

From what I can see, it is caused by the margins set by this selector:
`.android .intro .critter-animation, .ios .intro .critter-animation`

Removing that solves the problem.

schalkneethling commented:
![screen shot 2015-10-28 at 16 45 28](https://cloud.githubusercontent.com/assets/10350960/10792500/910ea7fa-7d95-11e5-9d3c-622a4f41727f.png)

These icons seems to be ever so slightly off center.

schalkneethling commented:
@malena Do we not have high-res versions of the images? From what I can see in Fx, when loading the page on high-def, it still loads the same PNG files as it does for normal resolutions.

malena commented:
@schalkneethling I got svg images but had issues with supporting ie > 10 and ie < 9 at the same time. The original linear-gradient fix I had doesn't work on ie11 since I have to use the prefix -ms-linear-gradient but then the ordering of the rules made it either work in ie >10 but not in ie11. Anyways, I replaced them with PNG's because I couldn't find a good working solution. I can ask for high res versions to swap them out.

schalkneethling commented:
@malena If you have access to the original files, the high-res will just be double the size of the standard PNG. Then you can use the `at2x` mixin to load the different versions.

With that said, I will not hold up this PR while we wait for those images or, create them ourselves. If all the other items are addressed, we can merge this one and add the high-res images in a separate PR.

Thanks @malena

malena commented:
@schalkneethling okay sounds good! I can address images other PR. Can you check the android fix I just committed?

schalkneethling commented:
@malena The changes for ios and android are all good. The icons are still a bit off center. I had a look and adjusting a few of the rules as follows, seems to do the trick:

```
.when-to-use li:first-child p {
    background-position: 90px top;
}

.when-to-use li:first-child + li p {
    background-position: -180px top;
}

.when-to-use li p {
    transparent url("/media/img/firefox/private-browsing/sprite-features.png") no-repeat scroll -490px top / 743px auto;
}
```

I am also wondering if it would not be better to give each of those list element a class. Then you can avoid selectors such as `.when-to-use li:first-child + li p` for example and just have:

```
.when-to-use li.classname p
```

Or even, place the class on the `p`, and then have:

```
.when-to-use p.classname
```
Let me know what you think. This is super close. In fact, I believe this is the last item that needs to be addressed. Thanks!

schalkneethling commented:
The shield here at around the desktopBreakpoint is not completely centered.

![screen shot 2015-10-28 at 20 30 38](https://cloud.githubusercontent.com/assets/10350960/10798965/e94bccb4-7db2-11e5-8447-645eec280845.png)

I find that adjusting the margin as below, resolves the problem.

```
#tracking-protection-animation .shield-container {
    margin: 0 0 0 -25px
}
```
Give it a try and let me know what you think.

malena commented:
@schalkneethling I have updated the alignment. I left the html how it was and didn't add any class names. Just wanted to deal with the css for now.

schalkneethling commented:
w00t r+ :de: Thanks @malena, looking great! - Just needs a quick squash.

malena commented:
@schalkneethling wahoo!! squashed :dancer:
